pkg/virt-controller/services/template.go

```diff
@@ -1315,6 +1317,9 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, t
 		pod.Spec.ServiceAccountName = serviceAccountName
 		automount := true
 		pod.Spec.AutomountServiceAccountToken = &automount
+	} else if val, ok := vmi.GetAnnotations()[ISTIO_INJECT_ANNOTATION]; ok && strings.ToLower(val) == "true" {
```

Comment:
it reads well enough, but possibly exporting it to a tiny func - `hasIstioAnnotation` maybe - would make it better. Not a must - I think it reads well enough.